### DISCUSSION TITLE
Uptime Metric

### DISCUSSION THREAD

**Discussion by CobraDunn** [ESPresense/ESPresense#109] [Q&A]
I'm a recent convert from room-assistant. I have (10) D1 Mini's running at this point and am still getting my feet under me during the conversion.

In the HA entitiy attributes, the ESP Uptime metric seems to indicate that the ESPs are rebooting regularly. Sometimes within a few minutes, and sometimes a few hours. I haven't seen any show an Uptime of more than 15 hours. Am I looking at this correctly?

![image](https://user-images.githubusercontent.com/65230622/139561240-80a63ca6-20a5-469b-8448-a4f937e7f663.png)

**Comment by DTTerastar** [comment]
The reset reason of RtcWdtRtc does say something is getting stuck.  Can you hook up a long term serial port and log it for 18 hours so we can catch what it's doing?  (Attach it)

**Reply by RaoulSargent** [reply]
I see similar behaviour. 
Have 5x esp32 running v2.0.15

Only noticed this behaviour this week so please forgive the un-scientific debugging at this time.

They appear to random (in time) restart. 

Sometimes (after a couple of days) I have noticed that they go "offline" (LWT MQTT message) and off network. In this case I have to power cycle as cannot get to web page nor ping. 

Was thinking they were suffering a memory leak of some kind or seeing too many devices so I currently force restart every 1 hour (if they stay up that long) hoping this will prevent them locking up. 

A lockup (needing physical power cycle) is more problematic than a soft restart.  So trying a regular hourly restart to see if I can prevent the lockups.

**Reply by RaoulSargent** [reply]
Will now start logging the restart reason as mentioned in previous reply. 

Will obtain a trace also.
